shogun-lib/src/main/java/de/terrestris/shogun/lib/security/access/BasePermissionEvaluator.java

```diff
@@ -75,6 +71,31 @@ public boolean hasPermission(Authentication authentication, Object targetDomainO
         Optional<User> userOpt = userProviderService.getUserFromAuthentication(authentication);
         User user = userOpt.orElse(null);
 
+        if (targetDomainObject instanceof Class<?>) {
+            // test if user/group class permission for provided class exists
+            Class clazz = (Class<?>) targetDomainObject;
+            String className = ((Class<?>) targetDomainObject).getCanonicalName();
+            BaseEntityPermissionEvaluator entityPermissionEvaluator =
+                this.getPermissionEvaluatorForClass(className);
+            final PermissionType permission = PermissionType.valueOf((String) permissionObject);
+            return entityPermissionEvaluator.hasPermission(user, clazz, permission);
+        }
+
+        // todo: for this we need either
+        //  - targetDomainType as parameter
+        //  - a way to get an Entity from a InstancePermission
+//        if (targetDomainObject instanceof InstancePermission) {
+//            // test if permission for entity of basePermission exists
+//            Long entityId = ((InstancePermission) targetDomainObject).getEntityId();
+//            BaseEntityPermissionEvaluator entityPermissionEvaluator =
+//                this.getPermissionEvaluatorForClass(targetDomainType);
+//
+//            log.trace("Checking permissions with permission evaluator '{}'",
+//                entityPermissionEvaluator.getClass().getSimpleName());
+//
+//            return entityPermissionEvaluator.hasPermission(user, entityId, targetDomainType, permissionObject);
+//        }
```

Comment:
Should this comment be removed?

Reply:
As this is marked with a TODO we should keep it.